### PR TITLE
Polished TensorFlowVariables code and documentation

### PR DESCRIPTION
**Pull request by Wapaul1**

### CODE REVIEW

File: doc/source/using-ray-with-tensorflow.rst
Comment (pcmoritz) [inline doc/source/using-ray-with-tensorflow.rst]: in _the_ same TensorFlow

File: doc/source/using-ray-with-tensorflow.rst
Comment (pcmoritz) [inline doc/source/using-ray-with-tensorflow.rst]: which variables _to_ set

File: doc/source/using-ray-with-tensorflow.rst
Comment (pcmoritz) [inline doc/source/using-ray-with-tensorflow.rst]: One common issue arises when two networks are defined in the same TensorFlow graph. In this case, TensorFlow appends an underscore and integer to the names of variables to disambiguate them.

File: doc/source/using-ray-with-tensorflow.rst
Comment (pcmoritz) [inline doc/source/using-ray-with-tensorflow.rst]: get rid of "inside"

File: doc/source/using-ray-with-tensorflow.rst
Comment (pcmoritz) [inline doc/source/using-ray-with-tensorflow.rst]: replace "I" with "we" everywhere

File: doc/source/using-ray-with-tensorflow.rst
Comment (pcmoritz) [inline doc/source/using-ray-with-tensorflow.rst]: Another issue to keep in mind is that ``TensorFlowVariables`` needs to add new operations to the graph. If you close the graph...

File: python/ray/experimental/tfutils.py
Comment (pcmoritz) [inline python/ray/experimental/tfutils.py]: Returns a dictionary containing the weights of the network

File: python/ray/experimental/tfutils.py
Comment (pcmoritz) [inline python/ray/experimental/tfutils.py]: TensorFlow

File: python/ray/experimental/tfutils.py
Comment (pcmoritz) [inline python/ray/experimental/tfutils.py]: Possible cause: Two networks were defined in the same TensorFlow graph.

File: python/ray/experimental/tfutils.py
Comment (pcmoritz) [inline python/ray/experimental/tfutils.py]: or additional variables that are passed in

File: python/ray/experimental/tfutils.py
Comment (pcmoritz) [inline python/ray/experimental/tfutils.py]: Placeholders for the weights.

**Comment by richardliaw** [comment]
@pcmoritz @Wapaul1 are we going to merge these changes?

**Comment by robertnishihara** [comment]
@richardliaw probably should, what do you guys think?

**Comment by richardliaw** [comment]
Sounds good; let me add some comments.

File: python/ray/experimental/tfutils.py
Comment (richardliaw) [inline python/ray/experimental/tfutils.py]: nit: can do `assert assign_list, ("No Variables...`

File: test/tensorflow_test.py
Comment (richardliaw) [inline test/tensorflow_test.py]: Can you put this as a docstring inside the test function?

File: test/tensorflow_test.py
Comment (richardliaw) [inline test/tensorflow_test.py]: Can you put this as a docstring inside the function?

**Review by richardliaw — COMMENTED** [review]
lgtm after small comments and fix merge conflicts. @Wapaul1

File: python/ray/experimental/tfutils.py
Comment (richardliaw) [inline python/ray/experimental/tfutils.py]: @Wapaul1 can you fix the indentation to 4 spaces?

Comment (richardliaw) [reply python/ray/experimental/tfutils.py]: @Wapaul1 just bumping this

**Comment by robertnishihara** [comment]
@Wapaul1 looks like there are some linting errors.

**Comment by richardliaw** [comment]
this is still not passing lint, but I can fix it later..

**Comment by robertnishihara** [comment]
Thanks @richardliaw, I'm taking a quick look now.

**Comment by robertnishihara** [comment]
This looks good to me, however, I can't quite remember the exact use case we had in mind for this. @Wapaul1 or @richardliaw can you remind me? It'd be great to explain it in the PR description.

**Comment by richardliaw** [comment]
I think this was just cleaning up the docs - no particular necessity

**Comment by Wapaul1** [comment]
Documentation, updates to tests, and allowing users to add extra variables they want to set.

**Comment by robertnishihara** [comment]
Thanks @Wapaul1, I guess what I'm asking is when do users need/want to add extra variables? What applications/scenarios give rise to this?

**Comment by Wapaul1** [comment]
One scenario, and the primary reason I added this functionality, is if you are only training part of a network. If you plug only part of the network into TensorFlowVariables, new nodes will also be created for all of its dependencies via BFS, which would inflate the size of the weights needlessly. This does not occur for variables passed through the new argument.

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**
